code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,311 @@
+	var/potentialspawns = list(/mob/living/simple_animal/hostile/asteroid/elite/goliath)
+	icon = 'icons/mob/lavaland/elite_lavaland_monsters.dmi'
+	icon_state = "elite_crevice"
+	light_color = LIGHT_COLOR_RED
+	light_range = 3
+	anchored = TRUE
+	density = FALSE
+	
+/obj/structure/elite_crevice/attack_hand(mob/user)
+	switch(activity)
+		if(CREVICE_INACTIVE)
+			if(istype(user, /mob/living/carbon/human))
+				activity = CREVICE_ACTIVE
+				var/mob/dead/observer/elitemind = null
+				visible_message("<span class='boldwarning'>The crevice glows.  Your instincts tell you to step back.</span>")
+				activator = user
+				if(boosted)
+					visible_message("<span class='boldwarning'>Something within the crevice stirs...</span>")
+					var/list/candidates = pollCandidatesForMob("Do you want to play as a lavaland elite?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, src, POLL_IGNORE_SENTIENCE_POTION)
+					if(candidates.len)
+						audible_message("<span class='boldwarning'>The stirring sounds increase in volume!</span>")
+						elitemind = pick(candidates)
+						SEND_SOUND(elitemind, sound('sound/effects/magic.ogg'))
+						to_chat(elitemind, "<b>You have been chosen to play as a Lavaland Elite.\nIn a few seconds, you will be summoned on Lavaland as a monster to fight your activator, in a fight to the death.\nYour attacks can be switched using the buttons on the top left of the HUD, and used by clicking on targets or tiles similar to a gun.\nWhile the opponent might have an upper hand with  powerful mining equipment and tools, you have great power normally limited by AI mobs.\nIf you want to win, you'll have to use your powers in creative ways to ensure the kill.  It's suggested you try using them all as soon as possible.\nShould you win, you'll receive extra information regarding what to do after.  Good luck!</b>")
+						addtimer(CALLBACK(src, .proc/spawn_elite, elitemind), 100)
+						return
+					else
+						visible_message("<span class='boldwarning'>The stirring stops, and nothing emerges.  Perhaps try again later.</span>")
+						activity = CREVICE_INACTIVE
+						activator = null
+						return
+				else
+					addtimer(CALLBACK(src, .proc/spawn_elite), 30)
+				return
+		if(CREVICE_PASSIVE)
+			if(istype(user, /mob/living/carbon/human))
+				activity = CREVICE_ACTIVE
+				visible_message("<span class='boldwarning'>The crevice glows as your arm enters its radius.  Your instincts tell you to step back.</span>")
+				activator = user
+				INVOKE_ASYNC(src, .proc/arena_trap)
+				if(boosted)
+					SEND_SOUND(mychild, sound('sound/effects/magic.ogg'))
+					to_chat(mychild, "<b>Someone has activated your crevice.  You will be returned to fight shortly, get ready!</b>")
+				addtimer(CALLBACK(src, .proc/return_elite), 30)
+				return
+
+				
+obj/structure/elite_crevice/proc/spawn_elite(var/mob/dead/observer/elitemind)
+	var/selectedspawn = pick(potentialspawns)
+	mychild = new selectedspawn(loc, src)
```

Comment:
Try to avoid sending extra arguments in new(). You don't use the second one, so just remove it.